### PR TITLE
Inversion of control

### PR DESCRIPTION
**Pull request by arothuis**
[Ticket 112778263](https://www.pivotaltracker.com/story/show/112778263)

_Note: because of the cache and log dir changes, deploy should be updated._

### CODE REVIEW

**Comment by arothuis** [comment]
@DRvanR

File: src/OpenConext/EngineBlockBridge/Configuration/EngineBlockConfiguration.php [hunk 1]
```diff
@@ -52,4 +52,32 @@ public function get($path, $default = null)
...
+
+        return $result;
+    }
+
+    /**
+     * Magic function so that $obj->value and related expression language will work
+     * @param string $name
+     * @return mixed|null|EngineBlockConfiguration
+     */
+    public function __get($name)
```
Comment (DRvanR) [inline src/OpenConext/EngineBlockBridge/Configuration/EngineBlockConfiguration.php]: I'd really would like to have this method removed. Why not use the method call (see [the related docs][1] and the [service example][2])?

`"@=service('engineblock.bridge.config').logger.conf.handler.syslog.conf.formatter.conf.format"`
would simply be `"@=service('engineblock.bridge.config').get('logger.conf.handler.syslog.conf.formatter.conf.format')"`

[1]: http://symfony.com/doc/2.7/components/expression_language/syntax.html
[2]: http://symfony.com/doc/2.7/book/service_container.html#using-the-expression-language

File: src/OpenConext/EngineBlockBundle/OpenConextEngineBlockBundle.php [hunk 2]
```diff
@@ -2,8 +2,19 @@
...
 namespace OpenConext\EngineBlockBundle;
 
+use EngineBlock_ApplicationSingleton;
 use Symfony\Component\HttpKernel\Bundle\Bundle;
 
 class OpenConextEngineBlockBundle extends Bundle
 {
+    public function boot()
+    {
+        parent::boot();
```
Comment (DRvanR) [inline src/OpenConext/EngineBlockBundle/OpenConextEngineBlockBundle.php]: can be removed as that method is a noop

File: src/OpenConext/EngineBlockFunctionalTestingBundle/OpenConextEngineBlockFunctionalTestingBundle.php [hunk 3]
```diff
@@ -7,7 +7,8 @@
 
 class OpenConextEngineBlockFunctionalTestingBundle extends Bundle
 {
+    public function boot()
+    {
+        \SAML2_Compat_ContainerSingleton::setContainer(new Container());
```
Comment (DRvanR) [inline src/OpenConext/EngineBlockFunctionalTestingBundle/OpenConextEngineBlockFunctionalTestingBundle.php]: please import, and add a comment as to what this does and why this is done.

File: tests/bootstrap.php [hunk 4]
```diff
@@ -21,4 +21,4 @@
 $config->testing = true;
 $config->hostname = 'engine.demo.openconext.org';
 $application->setConfiguration($config);
-$application->bootstrap();
+$application->bootstrap(new Psr\Log\NullLogger(), 'requestId');
```
Comment (DRvanR) [inline tests/bootstrap.php]: few questions:

- is this file really needed?
- the logger is set a few lines above as well, shouldn't that be removed?
- the hostname does not make sense.

**Comment by DRvanR** [comment]
:+1:

**Comment by DRvanR** [comment]
:+1:
